tests/src/TestExecutorTimeout.cpp

```diff
@@ -0,0 +1,19 @@
+#include "libserver/Executor.hpp"
+
+namespace {
+
+void TestExecutorTimeout()
+{
+  const auto executor = new server::Executor(50);
+  executor->execute([]{
+    std::this_thread::sleep_for(std::chrono::milliseconds(500));
+  });
+  delete executor;
```

Comment:
This test has no assertions. It should check something to verify that the executor killed the task due to a timeout, for example, have the task sleep for 500ms and then flip a boolean and have the assertion check that the boolean hasnt been flipped